### PR TITLE
[FEAT] Combined repositories screen with the workflow screen

### PR DESCRIPTION
**Pull request by YASH-RAJ-HANS**
-> Pagination is working.
-> Repos are fetched in latest push manner 
-> all repos are fetched including cross owner and organization public one
-> Combined repositories screen with the workflow screen

image
image
image

image

### CODE REVIEW

File: server/src/controllers/github.controller.ts [hunk 1]
```diff
@@ -99,6 +105,7 @@ export const repositories = async (
     }));
 
     res.json(repositories);
+    console.log("repo body",response.data);
```
Comment (sahilarora02) [inline server/src/controllers/github.controller.ts]: Try to use logger now

Also use it while development ,remove them while pushing to git

(unless its important)

File: server/src/controllers/github.controller.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ export const repositories = async (
   const { Octokit } = await import("octokit");
   const accessToken = req.body.access_token;
   const userName = req.body.user_name;
+  const { type = "all", sort = "pushed", direction = "desc", per_page = 20, page = 1 } = req.body;
```
Comment (sahilarora02) [inline server/src/controllers/github.controller.ts]: why per_page is not 30??
i though 30 was default !!?

File: client/my-app/src/app/repositories/page.tsx [hunk 3]
```diff
@@ -1,110 +1,113 @@
...
   useEffect(() => {
-    const fetchRepos = async () => {
-      if (username === null || access_token === null) {
-        console.log("No access token or username found");
-        return;
-      }
-      // setAccessToken(access_token);
-      // setUser(username);
+    if (username && access_token) {
+      const fetchRepos = async () => {
```
Comment (sahilrohera10) [inline client/my-app/src/app/repositories/page.tsx]: this check should be on function call not on function definition

File: client/my-app/src/components/UserRepoTable.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            Last Updated
+          </TableHead>
+          <TableHead className="text-right w-1/12 text-zinc-100 mb-6">
+            Connection
+          </TableHead>
+        </TableRow>
+      </TableHeader>
+      <TableBody>
+        {repos.map((repo) => (
+          <TableRow
```
Comment (sahilrohera10) [inline client/my-app/src/components/UserRepoTable.tsx]: add repo interface here

File: client/my-app/src/components/Workflow.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+import { useEffect, useState } from 'react'
+import { Input } from "@/components/ui/input"
+import { Check } from 'lucide-react'
+import { Button } from "@/components/ui/button"
+import { Card } from "@/components/ui/card"
+import { Badge } from "@/components/ui/badge"
+
+interface WorkflowProps {
+  access_token: string | null;
+  repo_url: string | null;
```
Comment (sahilrohera10) [inline client/my-app/src/components/Workflow.tsx]: take access_token from local storage

dont pass repo_url as props .. take repo_url from url params

File: client/my-app/src/components/Workflow.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  const [showEmailInput, setShowEmailInput] = useState(false)
+  const [email, setEmail] = useState('')
+  const [isDeploying, setIsDeploying] = useState(false)
+  const [repoName, setRepoName] = useState<string | null>(null)
+  const [owner, setOwner] = useState<string | null>(null);
+  const baseUrl = process.env.NEXT_PUBLIC_BASE_URL
+  const user_id = localStorage.getItem('user_id') || ''
+
+  const repoNameFromUrl = new URLSearchParams(window.location.search).get('repoName')
+
```
Comment (sahilrohera10) [inline client/my-app/src/components/Workflow.tsx]: take repo_url in this way

File: client/my-app/src/components/Workflow.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      owner,
+      repo: repoName,
+      access_token,
+      additional_email: email,
+      repo_url,
+      user_id
+    }
+    console.log("Payload Body",payload);
+    console.log("Repo Url:",repo_url);
+    try {
```
Comment (sahilrohera10) [inline client/my-app/src/components/Workflow.tsx]: remove consoles

File: client/my-app/src/components/Workflow.tsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+      const { owner, repoName } = extractOwnerAndRepo(repoUrlFromQuery);
+      setOwner(owner);
+      setRepoName(repoName);
+    }
+    const token = localStorage.getItem("access_token");
+    if (token) {
+      setAccessToken(token);
+    }
+  }, [searchParams]);
+
```
Comment (sahilrohera10) [inline client/my-app/src/components/Workflow.tsx:50]: why searchParams is in the dependency array ..

rather " const repoUrlFromQuery = searchParams.get("repo"); " should come outside the useEffect and " repoUrlFromQuery " should come in the dependency array...

here is the reason bcz ... everytime when the repoUrlFromQuery changes then we need to run the useEffect

**Review by sahilrohera10 — CHANGES_REQUESTED** [review]

File: client/my-app/src/components/UserRepoTable.tsx (hunk 4 above)
Comment (YASH-RAJ-HANS) [reply client/my-app/src/components/UserRepoTable.tsx]: done

**Review by sahilrohera10 — APPROVED** [review]

**Pull request merged**